boa_engine/src/syntax/ast/declaration/variable.rs

```diff
@@ -155,6 +171,7 @@ impl ToInternedString for VariableList {
     }
 }
 
+/// The error returned by the [`VariableList::try_from`] function.
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub struct TryFromVariableListError(());
```

Comment:
Why does this have an empty tuple inside? (Not from this PR, but it might be easier to do just this, right?
```suggestion
pub struct TryFromVariableListError;
```

Reply:
It's just standard Rust practices to disallow creating instances of the error outside of the crate; without this, a simple change like adding a private field would be a breaking change, for example.